### PR TITLE
Adding an assert verifying that you don't use different classes unless you're using projections

### PR DESCRIPTION
**Pull request by plivesey**

### CODE REVIEW

File: ConsistencyManager/Protocols/ConsistencyManagerModel.swift [hunk 1]
```diff
@@ -232,7 +232,7 @@ public extension ConsistencyManagerModel {
         // Usually, we don't need to merge and instead just return the other model.
         // This is because when we're not using projections, classes of the same id will always be of the same type.
         // So, we should just replace the current model with the updated model.
-        // TODO: Should we assert that the classes are the same here?
+        assert(self.dynamicType == model.dynamicType, "Two models of different classes have the same ID. This is not allowed without override mergeModel(:). See the documentation for more information on projections.")
```
Comment (nicksnyder) [inline ConsistencyManager/Protocols/ConsistencyManagerModel.swift]: is it possible to add the string representation of those dynamic types to the assertion message? might make debugging easier if it happens

Comment (plivesey) [inline ConsistencyManager/Protocols/ConsistencyManagerModel.swift]: It is possible

**Comment by plivesey** [comment]
@nicksnyder updated

**Comment by nicksnyder** [comment]
Lgtm

**Pull request merged**